euphonic/cli/utils.py

```diff
@@ -61,3 +179,135 @@ def matplotlib_save_or_show(save_filename: str = None):
+    """Get information about band path labels and break points
+
+    Parameters
+    ----------
+    bandpath
+        Bandpath dictionary from Seekpath
+
+    Returns
+    -------
+    list[int]
+
+        Indices at which the spectrum should be split into subplots
+
+    """
+    # Find break points between continuous spectra: wherever there are two
+    # adjacent labels
+    labels = np.array(bandpath["explicit_kpoints_labels"])
+
+    special_point_bools = np.fromiter(
+        map(bool, labels), dtype=bool)
+
+    # [T F F T T F T] -> [F F T T F T] AND [T F F T T F] = [F F F T F F] -> 3,
+    adjacent_non_empty_labels = np.logical_and(special_point_bools[:-1],
+                                               special_point_bools[1:])
+
+    adjacent_different_labels = (labels[:-1] != labels[1:])
+
+    break_points = np.where(np.logical_and(adjacent_non_empty_labels,
+                                           adjacent_different_labels))[0]
+    return (break_points + 1).tolist()
+
+
+def _insert_gamma(bandpath: dict) -> None:
+    """Modify seekpath.get_explicit_k_path() results; duplicate Gamma
+
+    This enables LO-TO splitting to be included
+    """
+    import numpy as np
+    gamma_indices = np.where(np.array(bandpath['explicit_kpoints_labels'][1:-1]
+                                      ) == 'GAMMA')[0] + 1
+
+    rel_kpts = bandpath['explicit_kpoints_rel'].tolist()
+    labels = bandpath['explicit_kpoints_labels']
+    for i in reversed(gamma_indices.tolist()):
+        rel_kpts.insert(i, [0., 0., 0.])
+        labels.insert(i, 'GAMMA')
+
+    bandpath['explicit_kpoints_rel'] = np.array(rel_kpts)
+    bandpath['explicit_kpoints_labels'] = labels
+
+    # These unused properties have been invalidated: safer
+    # to leave None than incorrect values
+    bandpath['explicit_kpoints_abs'] = None
+    bandpath['explicit_kpoints_linearcoord'] = None
+    bandpath['explicit_segments'] = None
+
+
+XTickLabels = List[Tuple[int, str]]
+SplitArgs = Dict[str, Any]
+
+
+def _bands_from_force_constants(data: ForceConstants,
+                                q_distance: Quantity,
+                                insert_gamma=True
+                                ) -> Tuple[QpointPhononModes,
+                                           XTickLabels, SplitArgs]:
+    structure = data.crystal.to_spglib_cell()
+    bandpath = seekpath.get_explicit_k_path(
+        structure,
+        reference_distance=q_distance.to('1 / angstrom').magnitude)
+
+    if insert_gamma:
+        _insert_gamma(bandpath)
+
+    x_tick_labels = _get_tick_labels(bandpath)
+    split_args = {'indices': _get_break_points(bandpath)}
+
+    print(
+        "Computing phonon modes: {n_modes} modes across {n_qpts} q-points"
+        .format(n_modes=(data.crystal.n_atoms * 3),
+                n_qpts=len(bandpath["explicit_kpoints_rel"])))
+    qpts = bandpath["explicit_kpoints_rel"]
+    modes = data.calculate_qpoint_phonon_modes(qpts, reduce_qpts=False)
```

Comment:
I just thought, we should use `asr='reciprocal'` here as it will make the gamma-point acoustic modes look much nicer. With that addition I'm happy for this PR to be merged.

Reply:
I'm of the view that these corrections should always be optional. If the ASR is doing too much work I don't trust the rest of the frequencies... We can have an argument to control it?

Reply:
I guess the most natural thing would be an argument `--asr` which accepts values `'real'` or `'reciprocal'`, defaults to `None` and has `constant='reciprocal'` so that `euphonic-bandstructure --asr phonons.yaml` will perform our preferred correction. This argument can be directly passed to the `asr=` argument of `calculate_qpoint_phonon_modes()`.

If we are very keen to have ASR by default, then extra logic is needed to parse `none` to None. (Personally, I'd rather leave it explicit and teach people to use `--asr` by prominently featuring it in tutorials/examples.)

Reply:
An `--asr` option seems good to me!

Reply:
Done!

As an aside, the 'realspace' ASR correction looks hilariously bad on the quartz test data compared with the uncorrected case. Is there a bug there or is this some pathological case for the method?

Reply:
The 'realspace' ASR correction is applied to the force constants matrix, and the 'reciprocal' ASR correction is applied to the dynamical matrix. As Quartz requires a dipole correction be applied to the force-constants matrix, and the ASR correction is only applied the force constants matrix and not the dipole correction it makes sense it doesn't work for Quartz! The reciprocal correction works for more cases and has little performance impact so I think most of the time that will be the one to use